### DISCUSSION TITLE
USB enumeration name of latest firmware

### DISCUSSION THREAD

**Discussion by WhiteBookX** [ARMmbed/DAPLink#1064] [Q&A] [answered]
Hi, I am using the FRDM-K64F board with DAPLINK for education purposes. And I am not a professional embedded engineer.

There is a minor thing that bothers me: 
If I flash the DAPLink firmware (rev.0244) provided by the [OpenSDA page](https://www.nxp.com/design/design-center/software/sensor-toolbox/opensda-serial-and-debug-adapter:OPENSDA#FRDM-K64F), the board will enumerate it as its name "FRDM-K64F". 
If I flash the latest firmware (rev.0254) from the DAPLink release site, the board will enumerate it as DAPLINK. However, I prefer that the students see the board name in File Explorer, but I still want to use the latest DAPLink firmware with WebUSB interfaces. 

I am wondering if there is another latest firmware that I can download and flash to the board so that it can enumerate its name. If not, do I have to clone the DAPLink and compile it to change the name?

Thanks!

**Comment by mathias-arm** [comment, marked as answer]
It was something that was only fixed half a year ago (see changes in `source/board/frdmk64f.c` in 2ba8cd1, which was part of #1038). There has not been a release since them. So you either have to build `main` yourself or you could try one of the latest GitHub Action [builds for main](https://github.com/ARMmbed/DAPLink/actions?query=branch%3Amain) (look at the "Artifacts" at the end of a build page).

**Reply by WhiteBookX** [reply]
Thank you so much! Your solution resolved my issue.
(I observed that there are updated versions of the code not yet available on the release site. I've never used GitHub actions before, so this is another valuable information as well!)
